mesonbuild/coredata.py

```diff
@@ -682,7 +682,9 @@ def merge_user_options(self, options):
                 if type(oldval) != type(value):
                     self.user_options[name] = value
 
-    def is_cross_build(self) -> bool:
+    def is_cross_build(self, when_building_for: MachineChoice = MachineChoice.HOST) -> bool:
+        if when_building_for == MachineChoice.BUILD:
+            return False
```

Comment:
This is not correct. This will break, f.ex., building with MSVC and targeting x64-UWP when building on an x64 machine. The machine architecture is not the final say on whether we're cross compiling.

Reply:
@nirbheek This is when building a binary for the build machine, not the host machine.

Reply:
Specifically, in the UWP case, the target machine is x64, but the compiler is a cross-compiler (you cannot run its output because it depends on libraries that are not available), and the build is a cross build (for the same reason).

Reply:
I thought targets that produce binaries for the *build* machine would always produce outputs that the build machine can run? Such as a tool for generating code, which could be used to compile binaries for the host. (Which obviously the build machine may not be able to run.) Either we're talking about different things, or I'm missing something.

Reply:
@nirbheek the two choices here are "build" and "host". I think this is correct and we are just getting hung up on the naming.

Reply:
In particular, there is absolutely no comparing of platforms and what-not here. Basically the logic is "host machine + have cross file => is cross". You could have the native and cross files be identical and it would still say is cross.

Reply:
Ahh, I misunderstood what this code is doing, my bad. Also my second comment was posted before I saw your reply to my comment, sorry for the confusion. Github didn't update with your comment.